src/treetensornetworks/solvers/alternating_update.jl

```diff
@@ -1,62 +1,17 @@
 
-function _extend_sweeps_param(param, nsweeps)
-  if param isa Number
-    eparam = fill(param, nsweeps)
-  else
-    length(param) >= nsweeps && return param[1:nsweeps]
-    eparam = Vector(undef, nsweeps)
-    eparam[1:length(param)] = param
-    eparam[(length(param) + 1):end] .= param[end]
-  end
-  return eparam
-end
-
-function process_sweeps(
-  nsweeps;
-  cutoff=fill(1E-16, nsweeps),
-  maxdim=fill(typemax(Int), nsweeps),
-  mindim=fill(1, nsweeps),
-  noise=fill(0.0, nsweeps),
-  kwargs...,
-)
-  maxdim = _extend_sweeps_param(maxdim, nsweeps)
-  mindim = _extend_sweeps_param(mindim, nsweeps)
-  cutoff = _extend_sweeps_param(cutoff, nsweeps)
-  noise = _extend_sweeps_param(noise, nsweeps)
-  return maxdim, mindim, cutoff, noise, kwargs
-end
-
-function sweep_printer(; outputlevel, state, which_sweep, sw_time)
-  if outputlevel >= 1
-    print("After sweep ", which_sweep, ":")
-    print(" maxlinkdim=", maxlinkdim(state))
-    print(" cpu_time=", round(sw_time; digits=3))
-    println()
-    flush(stdout)
-  end
-end
-
 function alternating_update(
-  updater,
   projected_operator,
   init_state::AbstractTTN;
-  checkdone=(; kws...) -> false,
-  outputlevel::Integer=0,
-  nsweeps::Integer=1,
+  sweep_plans,  #this is really the only one beig pass all the way down
+  outputlevel, # we probably want to extract this one indeed for passing to observer etc.
+  checkdone=(; kws...) -> false,  ### move outside
   (sweep_observer!)=observer(),
   sweep_printer=sweep_printer,
-  write_when_maxdim_exceeds::Union{Int,Nothing}=nothing,
-  updater_kwargs,
-  kwargs...,
+  write_when_maxdim_exceeds::Union{Int,Nothing}=nothing, ### move outside
 )
-  maxdim, mindim, cutoff, noise, kwargs = process_sweeps(nsweeps; kwargs...)
-
   state = copy(init_state)
-
-  insert_function!(sweep_observer!, "sweep_printer" => sweep_printer) # FIX THIS
-
-  for which_sweep in 1:nsweeps
-    if !isnothing(write_when_maxdim_exceeds) &&
+  for (which_sweep, sweep_plan) in enumerate(sweep_plans)
```

Comment:
```suggestion
  for which_sweep in eachindex(sweep_plans)
    sweep_plan = sweep_plans[which_sweep]
```
I'm not a fan of `enumerate`, it doesn't necessarily give a valid index into the collection:
```julia
julia> using OffsetArrays

julia> collect(enumerate(OffsetVector(randn(10), -1)))
10-element Vector{Tuple{Int64, Float64}}:
 (1, 1.2097571700024081)
 (2, 1.9625931511876364)
 (3, -2.048843359679102)
 (4, -0.7910034346777152)
 (5, -1.7837242897474845)
 (6, 0.03179938683614588)
 (7, 0.4589608248671337)
 (8, 0.5987573798836898)
 (9, 0.3850021806565198)
 (10, 0.8128603681636092)

julia> collect(eachindex(OffsetVector(randn(10), -1)))
10-element Vector{Int64}:
 0
 1
 2
 3
 4
 5
 6
 7
 8
 9
```